### PR TITLE
e2e tst: add actions to create github teams

### PR DESCRIPTION
**Pull request by burmudar**
add actions to create github teams and assign users to those teams. Also
added actions to remove added teams.

## Test plan
tested in wb/e2e/poc

### CODE REVIEW

**Comment by burmudar** [comment]
Current dependencies on/for this PR:
* main
  * **PR #56774** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/56774" target="_blank">Graphite</a> 
    * **PR #56775** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/56775" target="_blank">Graphite</a> 
      * **PR #56776** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/56776" target="_blank">Graphite</a> 
        * **PR #56777** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/56777" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/56777?utm_source=stack-comment).

File: dev/tst/github_client.go [hunk 1]
```diff
@@ -93,3 +93,50 @@ func (gh *GitHubClient) DeleteUser(ctx context.Context, username string) error {
 
 	return nil
 }
+
+func (gh *GitHubClient) GetOrCreateTeam(ctx context.Context, org *github.Organization, name string) (*github.Team, error) {
+	team, resp, err := gh.c.Teams.GetTeamBySlug(ctx, org.GetLogin(), name)
```
Comment (jhchabran) [inline dev/tst/github_client.go:98]: Same thing as in the previous PR, regarding the errors

File: dev/tst/store.go [hunk 2]
```diff
@@ -92,18 +92,29 @@ func (s *ScenarioStore) GetScenarioUser(u GitHubScenarioUser) (*github.User, err
 
 func (s *ScenarioStore) SetTeam(gt *GitHubScenarioTeam, t *github.Team) {
 	s.T.Helper()
+	// Store it twice so we have two ways of retrieving a team
```
Comment (jhchabran) [inline dev/tst/store.go:95]: I'm not sure to understand why we need this

**Review by jhchabran — CHANGES_REQUESTED** [review]

File: dev/codehost_testing/team.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+// Get returns the corresponding GitHub Team object that was created by the `CreateTeam`
+//
+// This method will only return a Team if the Scenario that created it has been applied otherwise
+// it will panic.
+func (team *Team) Get(ctx context.Context) (*github.Team, error) {
+	if team.s.IsApplied() {
+		return team.get(ctx)
+	}
+	panic("cannot retrieve org before scenario is applied")
```
Comment (jhchabran) [inline dev/codehost_testing/team.go:31]: `t.Fatal` ?

**Review by jhchabran — APPROVED** [review]

**Pull request merged**